errors/common.go

```diff
@@ -209,3 +209,7 @@ func ErrModifyChainID() error {
 func IsModifyChainIDErr(err error) bool {
 	return IsSameError(errModifyChainID, err)
 }
+
+func WithType(err error, obj interface{}) error {
+	return Wrap(err, fmt.Sprintf("%T", obj))
```

Comment:
pp: good idea to extract common functionality.  Maybe you could rename this to `WithTypeDescr`

Reply:
@alpe good point, not sure that I like the verbosity of it though as it's kind of straightforward that type here refers to description, basically we append the type name to a parent error.

I propose to keep it the same for now, we can always refactor in one operation later if it still feels confusing.